File: billing_ui_app/src/app/core/models/billing/add-or-edit-adjustment-model.ts
import { UserInfo } from "./get-claim-by-identifier";

export class AddOrEditAdjustmentModelWithUserInfo extends UserInfo {
    serviceLineId: number;
    claimId: number;
    adjustmentDetails: AdjustmentDetailsModel[];
}

export class AdjustmentDetailsModel {
    adjustmentId?: number;
    amount: number;
    isPositive:boolean;
    groupCode: string;
    reasonCode: string;
}

File: Microservices/BillingService/BillingService.Domain/Models/PaymentClaimServiceLine/AddOrEditAdjustmentModel.cs
using System.Collections.Generic;

namespace BillingService.Domain.Models.PaymentClaimServiceLine
{
    public class AddOrEditAdjustmentModel : UserInfo
    {
        public int ClaimId { get; set; }
        public int ServiceLineId { get; set; }
        public List<AdjustmentDetailsModel> AdjustmentDetails { get; set; }
    }
    public class AdjustmentDetailsModel
    {
        public int? AdjustmentId { get; set; }
        public decimal Amount { get; set; }
        public bool? isPositive { get; set; }
        public string GroupCode { get; set; }
        public string ReasonCode { get; set; }
    }

    public class AddOrEditAdjustmentModelForBulkPosting : AddOrEditAdjustmentModel
    {
        public decimal? AllowedAmount { get; set; }
        public decimal PaymentAmount { get; set; }
    }
}
